ci-operator/step-registry/p2p/cnv-post-upgrade/p2p-cnv-post-upgrade-commands.sh

```
@@ -0,0 +1,583 @@
+        phase=$(${KUBECTL_CMD} get vm ${VM_NAME} -n ${NAMESPACE} -o jsonpath='{.status.phase}' 2>/dev/null || echo "")
+        log_info "VM ${VM_NAME} exists with phase: ${phase}"
+        return 0
+    else
+        log_warning "VM ${VM_NAME} does not exist"
+        return 1
+    fi
+}
+
+wait_for_vm_running() {
+    log_info "Waiting for VM ${VM_NAME} to reach Running state (timeout: ${TIMEOUT}s)..."
+
+    local elapsed=0
+    local interval=5
+
+    while [ ${elapsed} -lt ${TIMEOUT} ]; do
+        local phase
+        phase=$(${KUBECTL_CMD} get vmi ${VM_NAME} -n ${NAMESPACE} -o jsonpath='{.status.phase}' 2>/dev/null || echo "")
+
+        case ${phase} in
+            "Running")
+                log_success "VM ${VM_NAME} is now Running"
+                return 0
+                ;;
+            "Failed")
+                log_error "VM ${VM_NAME} failed to start"
+                show_vm_details
+                return 1
+                ;;
+            "Succeeded")
+                log_warning "VM ${VM_NAME} completed execution"
+                return 1
+                ;;
+            "")
+                log_info "VM ${VM_NAME} not found yet..."
+                ;;
+            *)
+                log_info "VM ${VM_NAME} current phase: ${phase}"
+                ;;
+        esac
+
+        sleep ${interval}
+        elapsed=$((elapsed + interval))
+    done
+
+    log_error "Timeout waiting for VM ${VM_NAME} to reach Running state"
+    return 1
+}
+
+wait_for_pod_ready() {
+    log_info "Waiting for launcher pod to be ready..."
+
+    local elapsed=0
+    local interval=5
+    local pod_name=""
+
+    # Wait for launcher pod to be created
+    while [ ${elapsed} -lt ${TIMEOUT} ]; do
+        # Find the launcher pod by checking ownerReferences
+        pod_name=$(${KUBECTL_CMD} get pods -n ${NAMESPACE} -l kubevirt.io=virt-launcher -o jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="'${VM_NAME}'")].metadata.name}' 2>/dev/null || echo "")
```

Always protect the shell / env. var. using double quote.
```shell
 pod_name=$("${KUBECTL_CMD}" get pods -n "${NAMESPACE}" -l kubevirt.io=virt-launcher -o jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="'"${VM_NAME}"'")].metadata.name}' 
```
Do not silence the error unless there is a good reason.
Let failure stop the test.

---

This a PQE script I have only helped to create the step because original script is in gitlab, because of time constraints I don't intent make any changes that are not necessary

My guess here is these pods take time to show up so it's not a good idea fail this instantly

---

This can go to improvement, except removing the ` 2>/dev/null || echo "` removal, that need to go in this PR.